modern/src/LanguageProvider.js

```diff
@@ -0,0 +1,18 @@
+import {LanguageContext} from "./LanguageContext";
+import React, {useState} from "react";
+import {defaultLanguage, setSelectedLanguage} from "./common/localization";
+
+const LanguageProvider = (props) => {
```

Comment:
I think we should combine `LanguageContext` and `LanguageProvider` into a single file.

It would also be nice to migrate existing "localization" code here, so we have a single place for all language related stuff.

Reply:
LanguageContext and LanguageProvider have been combined into a single file.

Are you thinking about migrating the existing code from the "localization.js" file into LanguageProvider and use react context for all localization related stuff?

Reply:
Yes, I don't want to have two different places to manage localization. I think all the code from `localization.js` should be integrated into `LanguageProvider`. And I mean not just copy, but integrate into the component.